esphome/components/ld2420/ld2420.h

```diff
@@ -0,0 +1,194 @@
+                             uint32_t rg14_move, uint32_t rg14_still, uint32_t rg15_move, uint32_t rg15_still) {
+    this->new_config_.high_thresh[(uint8_t) 0] = rg0_move;
+    this->new_config_.low_thresh[(uint8_t) 0] = rg0_still;
+    this->new_config_.high_thresh[(uint8_t) 1] = rg1_move;
+    this->new_config_.low_thresh[(uint8_t) 1] = rg1_still;
+    this->new_config_.high_thresh[(uint8_t) 2] = rg2_move;
+    this->new_config_.low_thresh[(uint8_t) 2] = rg2_still;
+    this->new_config_.high_thresh[(uint8_t) 3] = rg3_move;
+    this->new_config_.low_thresh[(uint8_t) 3] = rg3_still;
+    this->new_config_.high_thresh[(uint8_t) 4] = rg4_move;
+    this->new_config_.low_thresh[(uint8_t) 4] = rg4_still;
+    this->new_config_.high_thresh[(uint8_t) 5] = rg5_move;
+    this->new_config_.low_thresh[(uint8_t) 5] = rg5_still;
+    this->new_config_.high_thresh[(uint8_t) 6] = rg6_move;
+    this->new_config_.low_thresh[(uint8_t) 6] = rg6_still;
+    this->new_config_.high_thresh[(uint8_t) 7] = rg7_move;
+    this->new_config_.low_thresh[(uint8_t) 7] = rg7_still;
+    this->new_config_.high_thresh[(uint8_t) 8] = rg8_move;
+    this->new_config_.low_thresh[(uint8_t) 8] = rg8_still;
+    this->new_config_.high_thresh[(uint8_t) 9] = rg9_move;
+    this->new_config_.low_thresh[(uint8_t) 9] = rg9_still;
+    this->new_config_.high_thresh[(uint8_t) 10] = rg10_move;
+    this->new_config_.low_thresh[(uint8_t) 10] = rg10_still;
+    this->new_config_.high_thresh[(uint8_t) 11] = rg11_move;
+    this->new_config_.low_thresh[(uint8_t) 11] = rg11_still;
+    this->new_config_.high_thresh[(uint8_t) 12] = rg12_move;
+    this->new_config_.low_thresh[(uint8_t) 12] = rg12_still;
+    this->new_config_.high_thresh[(uint8_t) 13] = rg13_move;
+    this->new_config_.low_thresh[(uint8_t) 13] = rg13_still;
+    this->new_config_.high_thresh[(uint8_t) 14] = rg14_move;
+    this->new_config_.low_thresh[(uint8_t) 14] = rg14_still;
+    this->new_config_.high_thresh[(uint8_t) 15] = rg15_move;
+    this->new_config_.low_thresh[(uint8_t) 15] = rg15_still;
+  };
+
+  int32_t last_periodic_millis = millis();
+  int32_t last_normal_periodic_millis = millis();
+
+ protected:
+  struct RegConfigT {
+    uint16_t min_gate;
+    uint16_t max_gate;
+    uint16_t timeout;
+    uint32_t high_thresh[16];
+    uint32_t low_thresh[16];
+  };
+
+  struct CmdReplyT {
+    uint8_t command;
+    uint8_t status;
+    uint32_t data[4];
+    uint8_t length;
+    uint16_t error;
+    volatile bool ack;
+  };
+
+  RegConfigT current_config_;
+  RegConfigT new_config_;
+  CmdReplyT cmd_reply_;
+
+  void get_firmware_version_();
+  void set_min_max_distances_timeout_(uint32_t max_gate_distance, uint32_t min_gate_distance, uint32_t timeout);
+  int get_gate_threshold_(uint8_t gate);
+  void set_gate_threshold_(uint8_t gate);
+  void get_reg_value_(uint16_t reg);
+  void set_reg_value_(uint16_t reg, uint16_t value);
+  int get_min_max_distances_timeout_();
+  u_int8_t set_config_mode_(bool enable);
```

Comment:
Well this is weird, haha
```suggestion
  uint8_t set_config_mode_(bool enable);
```

Reply:
Isn't it. Compiler must know something we don't.